### ISSUE TITLE
mongodbatlas_private_endpoint and mongodbatlas_private_endpoint_interface_link not working as expected in version 0.7

### ISSUE THREAD

**Issue by AntjeRadkiewicz1337** [mongodb/terraform-provider-mongodbatlas#406]
Hi 😊

So, we just found out the hard way that there was a change for the mongodbatlas_private_endpoint and the mongodbatlas_private_endpoint_interface_link.

### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v0.13.6
registry.terraform.io/mongodb/mongodbatlas v0.7.0
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_private_endpoint" "mongo_vpc_endpoint" {
  project_id    = data.mongodbatlas_project.atlas_project.id
  provider_name = "AWS"
  region        = "censored"
}

resource "mongodbatlas_private_endpoint_interface_link" "mongo_vpc_endpoint_interface_link" {
  project_id            = mongodbatlas_private_endpoint.mongo_vpc_endpoint[0].project_id
  private_link_id       = mongodbatlas_private_endpoint.mongo_vpc_endpoint[0].private_link_id
  interface_endpoint_id = aws_vpc_endpoint.aws_vpc_endpoint.id
}

```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

### Expected Behavior
The version 0.7.0 should not replace the resources mongodbatlas_private_endpoint, mongodbatlas_private_endpoint_interface_link, aws_vpc_endpoint on each run and deprecation warnings should be printed out.

OR

Terraform should print an error and stop.

### Actual Behavior
Our pipeline destroyed our mongodbatlas_private_endpoint, mongodbatlas_private_endpoint_interface_link and aws_vpc_endpoint and created new ones and all our services couldn’t reach the databases for about 10 minutes.

```
mongodbatlas_private_endpoint.mongo_vpc_endpoint[0] must be replaced
~ endpoint_service_name = "censored" -> (known after apply)
      + error_message         = (known after apply)
      ~ id                    = "censored" -> (known after apply)
      ~ interface_endpoints   = [
          - "censored",
        ] -> (known after apply)
      ~ private_link_id       = "censored" -> (known after apply)
        project_id            = "censored"
      + provider_name         = "AWS" # forces replacement
      + region                = "censored" # forces replacement
      ~ status                = "AVAILABLE" -> (known after apply)

mongodbatlas_private_endpoint_interface_link.mongo_vpc_endpoint_interface_link must be replaced
resource "mongodbatlas_private_endpoint_interface_link" "mongo_vpc_endpoint_interface_link" {
      ~ connection_status     = "AVAILABLE" -> (known after apply)
      ~ delete_requested      = false -> (known after apply)
      + error_message         = (known after apply)
      ~ id                    = "censored" -> (known after apply)
      ~ interface_endpoint_id = "censored" -> (known after apply) # forces replacement
      ~ private_link_id       = "censored" -> (known after apply) # forces replacement
        project_id            = “censored"
    }

```

### Additional Context
As we were looking into the issue, we found out that the resources will be deprecated and there wouldn’t be a breaking change, but the deprecation warning is not printed with version 0.7.0. By trying to raise the version to 0.8.x we saw that the same changes would be applied but this time deprecation warnings were printed out.

We will now have to use the new resources, but obviously as they are not present in version 0.7.0 we will also have to migrate directly to version 0.8.x.

**Comment by themantissa** [comment]
@AntjeRadkiewicz1337 thank you for the issue, we'll take a look - looks like a potential bug 0.7.0 and not related to the privatelink changes (nothing changed for 0.7.0)  @nikhil-mongo can you take a look here?

**Comment by nikhil-mongo** [comment]
@AntjeRadkiewicz1337 
I tried to repro this but as you mentioned that it tried to update the privatelink resources and recreated them. 

- Can we know if something was changed in the configuration?
- The example in the repository is since the time v0.12 of terraform was out and possibly the mongodb atlas provider version 0.6.
- The recreate is possible only when there is change in the resource name 

I deployed the private link using MongoDB Atlas provider version 0.6 and then upgraded to version 0.7 and terraform reported no change.

**UPDATE** - The MongoDB Atlas provider version 0.7 does not recognizes the newly introduced resource types for privatelink. Therefore, there is no chance of the resource re-creation.
However when upgrading to the provider version 0.8.0 and updating the resources, this is a breaking change.

However, yes when changing to version 0.8, this should be taken care of.

**Comment by themantissa** [comment]
Just to update here - @nikhil-mongo is working to create an internal ticket so our developers can investigate the issue further.   Will update on progress here.

**Comment by nikhil-mongo** [comment]
PRODTRIAGE-1435 filed internally.

**Comment by AntjeRadkiewicz1337** [comment]
Sorry for the late answer.

It seems we didn't pin our version directely to "= 0.7" but to ">= 0.7" therefore our pipeline used the 0.8 version and recreated all the resources. After that we pinned the version to 0.7 but it always recreated our resources.

My suggestion would be that something in the terraform state was not matching anymore. We solved the problem for us by pinning the version now correctly and switching to version 0.8.2 (which runs smoothly ;) )

further information:
- we did not change or rename any configuration in this place.

**Comment by themantissa** [comment]
Corrected with release of 0.8.2 so closing.

**Comment by igoratencompass** [comment]
Further more for people that end up here looking for similar problem, seems that `v0.7` does not know how to properly replace the `mongodbatlas_private_endpoint` resource when you actually need it to. In the plan the action created is `+/-` like so:

```
   # module.mongodb-atlas["mylink"].mongodbatlas_private_endpoint.atlaspl must be replaced
+/- resource "mongodbatlas_private_endpoint" "atlaspl" {
     }
```
which results in a conflict of course since the endpoint already exists:

```
Error: error creating MongoDB Private Endpoints Connection: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/xxxxxxxxxxxxxxxxxxxx/privateEndpoint: 409 (request "Conflict") A PrivateLink Endpoint Service already exists for AWS region EU_WEST_2.
```
The action should be `-/+` instead so the link gets destroyed first and then created again.
